src/v/cluster/partition_probe.h

```diff
@@ -138,6 +171,11 @@ class replicated_partition_probe : public partition_probe::impl {
     int64_t _iceberg_commit_offset_lag{metric_default_initialized_state};
     metrics::internal_metric_groups _metrics;
     metrics::public_metric_groups _public_metrics;
+
+    bool _metrics_setup_pending{true};
```

Comment:
nit: would appreciate a comment about what this means, and what is or isn't allowed while setup is pending